### PR TITLE
refactor: separating query from submitting txs for client

### PR DESCRIPTION
**Pull request by SebastianElvis**
Replaces https://github.com/babylonchain/rpc-client/pull/24

https://github.com/babylonchain/rpc-client/pull/24 was mistakenly merged without squashing. This PR does it again over the previous head resetted by @gitferry

### CODE REVIEW

**Review by gitferry — APPROVED** [review]
Thanks!

File: go.sum [hunk 1]
```diff
@@ -227,20 +337,36 @@ github.com/facebookgo/ensure v0.0.0-20200202191622-63f1cf65ac4c h1:8ISkoahWXwZR4
...
 github.com/fsnotify/fsnotify v1.4.9/go.mod h1:znqG4EE+3YCdAaPaxE2ZRY/06pZUdp0tY4IgpuI1SZQ=
 github.com/fsnotify/fsnotify v1.5.4 h1:jRbGcIw6P2Meqdwuo0H1p6JVLbL5DHKAKlYndzMwVZI=
 github.com/fsnotify/fsnotify v1.5.4/go.mod h1:OVB6XrOHzAwXMpEM7uPOzcehqUV2UqJxmVXmkdnm1bU=
+github.com/gballet/go-libpcsclite v0.0.0-20190607065134-2772fd86a8ff/go.mod h1:x7DCsMOv1taUwEWCzT4cmDeAkigA5/QCwUodaVOe8Ww=
+github.com/getkin/kin-openapi v0.53.0/go.mod h1:7Yn5whZr5kJi6t+kShccXS8ae1APpYTW6yheSwk8Yi4=
+github.com/getkin/kin-openapi v0.61.0/go.mod h1:7Yn5whZr5kJi6t+kShccXS8ae1APpYTW6yheSwk8Yi4=
 github.com/ghodss/yaml v1.0.0/go.mod h1:4dBDuWmgqj2HViK6kFavaiC9ZROes6MMH2rRYeMEF04=
+github.com/gin-contrib/sse v0.1.0 h1:Y/yl/+YNO8GZSjAhjMsSuLt29uWRFHdHYUb5lYOV9qE=
+github.com/gin-contrib/sse v0.1.0/go.mod h1:RHrZQHXnP2xjPF+u1gW/2HnVO7nvIa9PG3Gm+fLHvGI=
+github.com/gin-gonic/gin v1.7.0 h1:jGB9xAJQ12AIGNB4HguylppmDK1Am9ppF7XnGXXJuoU=
```
Comment (gitferry) [inline go.sum:359]: According to a previous alert, this version might be malicious. Maybe we need to update it to `v1.7.7`

Comment (SebastianElvis) [reply go.sum:359]: It seems that Cosmos SDK v0.46.6 [is still using github.com/gin-gonic/gin v1.7.7](https://github.com/cosmos/cosmos-sdk/blob/v0.46.6/go.mod). So probably let's bump Cosmos SDK dependency and this one for every module in separate PRs. Wdyt?

Comment (gitferry) [reply go.sum:359]: I see. It makes sense. Thanks for the investigation!

**Pull request merged**